core/query_factory.py

```diff
@@ -269,3 +269,115 @@ def _make_for_test(self):
         query = self.make()
         query = query.replace(SPACE_INDENT, '').replace('\n', '')
         return query
+
+
+    def friendly_message(self):
+        """Create a friendly/human message about what the query will do.
+
+        @return: The message
+        @rtype: str
+        """
+        
+        self._check_parameters()
+
+        place = self._area
+        if self._area is not None:
+            # human format a list
+            if len(self._area) == 1:
+                place = ' ' + self._area[0] + ' '  # simply unwrap the list
+            elif len(self._area) == 2:
+                place = ' {} '.format(tr('and')).join(self._area)
+            else:
+                place = ', '.join(self._area[:-2]) + ", "
+                place = place + ' {} '.format(tr('and')).join(self._area[-2:])
+
+        # These are the text templates to be used
+        ALL_VALUES = tr("All OSM objects with the key {key} in {extent} are going to be downloaded")
+        ALL_VALUES_WITH_DIST = tr("All OSM objects with the key {key} in {dist} meters of {extent} are going to be downloaded")
+
+        NO_KEY = tr("All OSM objects in {extent} are going to be downloaded")
+        NO_KEY_WITH_DIST = tr("All OSM objects in {dist} meters of {extent} are going to be downloaded")
+
+        ATTRIB_ONLY = tr("All OSM objects with the key {key} are going to be downloaded")
+        
+        extent_lbl = ""
+        dist_lbl = ""
+        key_lbl = ""
+        use_with_dist = False
+        use_all_vals = False
+        attrib_only = False
+        
+        # first translate the location information
+        have_loc_info = False
+        if self._query_type == QueryType.InArea:  #'in':
+            if place is not None:
+                extent_lbl = place
```

Comment:
Is this code reachable?

The `check_parameters` has been called. So at this points, all checks should have been done. An exception is raised if the query is not good.